UnityProject/Assets/Scripts/Core/SecureStuff/SecureMapsSaver.cs

```diff
@@ -68,81 +87,557 @@ public virtual void AddID(string ToAdd)
-						if (attribute.Length > 0)
+						var PrefabComponent = JsonConvert.DeserializeObject<PrefabComponent>(ModField.Data);
+						try
 						{
-							continue;
+							var Prefab = IPopulateIDRelation.ObjectsFromForeverID(PrefabComponent.ForeverId,
+								Field.FieldType.GetGenericArguments()[0]);
+							List[Index] = ((GameObject) Prefab).GetComponent(PrefabComponent.ComponentName);
+						}
+						catch (Exception e)
+						{
+							Loggy.LogError(e.ToString());
 						}
 					}
-					else if (Field.IsPublic)
+				}
+				else
+				{
+					if (AllLoaded)
 					{
-						if (Field.IsNotSerialized)
+						while (List.Count <= Index)
+							//TODO Could be exploited? well You could just have a map with a million objects so idk xD
 						{
-							continue;
+							List.Add(null);
+						}
+
+						List[Index] = GetComponentPath(ModField.Data, IPopulateIDRelation);
+					}
+					else
+					{
+						IPopulateIDRelation.FlagSaveKey(RootID, Root, ModField);
+					}
+				}
+			}
+			else if (typeof(ScriptableObject).IsAssignableFrom(Field.FieldType.GetGenericArguments()[0]) &&
+			         typeof(IHaveForeverID).IsAssignableFrom(Field.FieldType.GetGenericArguments()[0]))
+			{
+				if (ModField.Data == "#removed#")
+				{
+					List.Remove(Index);
+				}
+				else
+				{
+					if (List == null)
+					{
+						Loggy.LogError("0oh no...");
+					}
+					while (List.Count <= Index)
+						//TODO Could be exploited? well You could just have a map with a million objects so idk xD
+					{
+						List.Add(null);
+					}
+
+					var SO = IPopulateIDRelation.ObjectsFromForeverID(ModField.Data,
+						Field.FieldType.GetGenericArguments()[0]);
+					List[Index] = SO;
+				}
+			}
+			else
+			{
+				if (ModField.Data == "#removed#")
+				{
+					List.Remove(Index);
+				}
+				else
+				{
+					while (List.Count <= Index)
+						//TODO Could be exploited? well You could just have a map with a million objects so idk xD
+					{
+						List.Add(null);
+					}
+
+					List[Index] = Librarian.Page.DeSerialiseValue(ModField.Data,
+						Field.FieldType.GetGenericArguments()[0]);
+				}
+			}
+		}
+
+		private static GameObject GetGameObjectPath(string Id, IPopulateIDRelation IPopulateIDRelation)
+		{
+			if (Id == "MISSING")
+			{
+				Loggy.LogError("Map has missing references");
+				return null;
+			}
+
+			var IDPath = Id.Split("@");
+			var Object = IPopulateIDRelation.Objects[IDPath[0]];
+			//0,1
+			List<int> IDs = new List<int>();
+
+			if (IDPath[1].Contains(
+				    ",")) //Technically it always has 0 , but we can ignore it so don't remove it technically
+			{
+				IDs = IDPath[1].Split().Select(x => int.Parse(x)).ToList();
+				IDs.RemoveAt(0);
+			}
+
+
+			while (IDs.Count > 0)
+			{
+				Object = Object.transform.GetChild(IDs[0]).gameObject;
+				IDs.RemoveAt(0);
+			}
+
+			return Object;
+		}
+
+		private static Component GetComponentPath(string Id, IPopulateIDRelation IPopulateIDRelation)
```

Comment:
another giant function, split

Reply:
what you talking about it's only 10 lines?